app/src/main/java/com/sunasterisk/youtubebackground/screen/adapter/VideoAdapter.kt

```diff
@@ -0,0 +1,55 @@
+package com.sunasterisk.youtubebackground.screen.adapter
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.squareup.picasso.Picasso
+import com.sunasterisk.youtubebackground.R
+import com.sunasterisk.youtubebackground.data.model.Video
+import com.sunasterisk.youtubebackground.utils.VideoUtils
+import kotlinx.android.synthetic.main.single_video_item.view.*
+
+class VideoAdapter()
+    : RecyclerView.Adapter<VideoAdapter.ViewHolder>()
+{
+    private val listVideos by lazy { mutableListOf<Video>() }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VideoAdapter.ViewHolder {
+        return ViewHolder(
+            LayoutInflater.from(parent.context).inflate(R.layout.single_video_item, parent, false)
+        )
+    }
+
+    override fun getItemCount(): Int {
+        return listVideos.size
+    }
+
+    override fun onBindViewHolder(holder: VideoAdapter.ViewHolder, position: Int) {
+            holder.bindData(listVideos.get(position))
+    }
+
+    fun updateData(videos: MutableList<Video>?) {
```

Comment:
check trước khi đưa vào hàm, tham số để not-nullable
``` Kotlin
videos?.let {
     updateData(it) 
}
```
```suggestion
    fun updateData(videos: MutableList<Video>) {
```